src/main/java/org/jenkins/tools/test/hook/AnalysisPomExecutionHook.java

```diff
@@ -0,0 +1,32 @@
+package org.jenkins.tools.test.hook;
+
+import java.util.Map;
+import java.util.Set;
+import org.jenkins.tools.test.model.PomData;
+
+/**
+ * Custom execution hook for plugins whose parent is {@code org.jvnet.hudson.plugins:analysis-pom}.
+ * These plugins use Maven Failsafe Plugin in their test suites.
+ */
+public class AnalysisPomExecutionHook extends PluginWithFailsafeIntegrationTestsHook {
```

Comment:
We have no internal users of `PluginWithFailsafeIntegrationTestsHook` (nee `PluginWithFailsafeIntegrationTestsHook`)

And as implemented `PluginWithFailsafeIntegrationTestsHook` is incorrect as to be generic it should be executing `pre-integration-test` then `integration-test` and finally `post-integration-test`.

the `analysis-pom`'sUllis [configuration of failsafe](https://github.com/jenkinsci/analysis-pom-plugin/blob/eba7e94cb6b9fc1bc2af497359c4a378a7e337dd/pom.xml#L269-L287)  does not follow this generic nature as it is not spinning anything up.

Perhaps it is worth just removing `PluginWithFailsafeIntegrationTestsHook` and then inlining `getGoals` here in a way that is specific to Ulli's usage.
```suggestion
public class AnalysisPomExecutionHook extends PluginWithIntegrationTestsHook {
```